### PR TITLE
Tweak intra-cell spacing.

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/ConversationView/ConversationViewItem.m [hunk 1]
```diff
@@ -271,8 +271,21 @@ - (CGFloat)vSpacingWithPreviousLayoutItem:(ConversationViewItem *)previousLayout
...
+            return 2.f;
+        }
+    } else if (self.interaction.interactionType == OWSInteractionType_OutgoingMessage
+        && previousLayoutItem.interaction.interactionType == OWSInteractionType_OutgoingMessage) {
+        return 2.f;
+    }
+
     // TODO:
-    return 4.f;
+    return 10.f;
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/ConversationView/ConversationViewItem.m]: This probably doesn't capture everything but it's closer.

**Comment by charlesmchen-signal** [comment]
> Have you looked at this section of the Yap docs? It sounds like it's solving the same problem. I wonder if we could simplify our code by using their existing implementation for these kinds of changes.

Woah, this is wonderful.  I've applied this.  It's a critical change, so it bears significant testing, but I'm very optimistic.  Thanks!

PTAL @michaelkirk

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m [hunk 2]
```diff
@@ -3332,15 +3343,16 @@ - (void)uiDatabaseDidUpdate:(NSNotification *)notification
...
+        [self reloadViewItems];
         [self.collectionView reloadData];
         [self updateLastVisibleTimestamp];
         [self cleanUpUnreadIndicatorIfNecessary];
         return;
     }
 
     NSUInteger oldViewItemCount = self.viewItems.count;
-    NSMutableSet<NSNumber *> *rowsThatChangedSize = [[self reloadViewItems] mutableCopy];
+    [self reloadViewItems];
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: Significant simplication.

File: SignalServiceKit/src/Network/WebSockets/TSSocketManager.m [hunk 3]
```diff
@@ -526,7 +526,7 @@ - (void)makeRequest:(TSRequest *)request success:(TSSocketMessageSuccess)success
     NSError *error;
     BOOL wasScheduled = [self.websocket sendDataNoCopy:messageData error:&error];
     if (!wasScheduled || error) {
-        OWSProdLogAndFail(@"%@ could not serialize request JSON: %@", self.logTag, error);
+        OWSProdLogAndFail(@"%@ could not send socket request: %@", self.logTag, error);
```
Comment (charlesmchen-signal) [inline SignalServiceKit/src/Network/WebSockets/TSSocketManager.m]: Fixing unrelated logging.

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m [hunk 4]
```diff
@@ -3303,8 +3310,13 @@ - (void)uiDatabaseDidUpdate:(NSNotification *)notification
...
                 YapCollectionKey *collectionKey = rowChange.collectionKey;
                 if (collectionKey.key) {
-                    ConversationViewItem *viewItem = self.viewItemCache[collectionKey.key];
-                    [self reloadInteractionForViewItem:viewItem];
+                    ConversationViewItem *_Nullable viewItem = self.viewItemCache[collectionKey.key];
+                    if (viewItem) {
+                        [self reloadInteractionForViewItem:viewItem];
+                    }
+                } else if (rowChange.indexPath && rowChange.originalIndex < self.viewItems.count) {
+                    // Do nothing, this is a pseudo-update generated due to
```
Comment (michaelkirk-signal) [inline Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: nit: can you add an assert:

OWSAssert(rowChange.changes == YapDatabaseViewChangedDependency)

File: Signal/src/ViewControllers/ConversationView/ConversationViewItem.h [hunk 5]
```diff
@@ -61,12 +61,6 @@ NSString *NSStringForOWSMessageCellType(OWSMessageCellType cellType);
 @property (nonatomic) BOOL shouldHideAvatar;
 
 @property (nonatomic) NSInteger row;
```
[the remark this answers is not in the document]
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/ConversationView/ConversationViewItem.h]: Done.

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m [hunk 6]
```diff
@@ -3303,8 +3310,13 @@ - (void)uiDatabaseDidUpdate:(NSNotification *)notification
             case YapDatabaseViewChangeUpdate: {
                 YapCollectionKey *collectionKey = rowChange.collectionKey;
                 if (collectionKey.key) {
-                    ConversationViewItem *viewItem = self.viewItemCache[collectionKey.key];
-                    [self reloadInteractionForViewItem:viewItem];
+                    ConversationViewItem *_Nullable viewItem = self.viewItemCache[collectionKey.key];
+                    if (viewItem) {
```
[the remark this answers is not in the document]
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: This shouldn't be necessary. I've just made sure that we set `hasMalformedRowChange` if the view item can't be found.

[the remark this answers is not in the document]
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: Note that `hasMalformedRowChange` now causes an `OWSProdLogAndFail`.

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m (hunk 4 above)
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: Good idea, done.

**Pull request merged**